arrow-libs/core/arrow-core-test/src/commonMain/kotlin/arrow/core/test/generators/utils.kt

```diff
@@ -0,0 +1,30 @@
+package arrow.core.test.generators
+
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.intrinsics.COROUTINE_SUSPENDED
+import kotlin.coroutines.intrinsics.intercepted
+import kotlin.coroutines.intrinsics.suspendCoroutineUninterceptedOrReturn
+import kotlin.coroutines.startCoroutine
+import kotlinx.coroutines.Dispatchers
+
+public suspend fun Throwable.suspend(): Nothing =
+  suspendCoroutineUninterceptedOrReturn { cont ->
+    suspend { throw this }.startCoroutine(
+      Continuation(Dispatchers.Default) {
+        cont.intercepted().resumeWith(it)
+      }
+    )
+
+    COROUTINE_SUSPENDED
+  }
+
+public suspend fun <A> A.suspend(): A =
```

Comment:
How is this disambiguated from the specialization?

Reply:
This is disambiguated by the type. The concrete signature takes precedence for top-level functions.

```
fun <A> suspend(a: A): A = ..
fun suspend(throwable: Throwable): Nothing = ...
```